### DISCUSSION TITLE
400 error occurs when using the search function while on page 2 or later in pagination

### DISCUSSION THREAD

**Discussion by haruki-hiraiwa** [aminalaee/sqladmin#798] [Q&A]
If I use the search function while on page 2 or later in pagination, I get a 400 error.
I think this is probably due to the second page not existing in the search results.
![スクリーンショット 2024-08-08 10 48 29](https://github.com/user-attachments/assets/69a3232e-8e11-49d3-8f0a-c5159fce1d24)

I did not see this error with sqladmin version 0.16.1, but with version 0.17.0.
So I think the error is due to the version of the library.

↓No error
sqladmin = {extras = ["full"], version = "^0.16.1"}
↓Error
sqladmin = {version = "0.17.0", extras = ["full"]}

![スクリーンショット 2024-08-08 10 43 22](https://github.com/user-attachments/assets/43d96946-4dfb-4d10-9220-937fac11e728)

**Comment by aminalaee** [comment]
Hi, yeah you are right, might be related to https://github.com/aminalaee/sqladmin/pull/782
PRs are welcome to fix this

**Comment by rvic** [comment]
Never did PRs before. As working solution I had to override list() method of ModelView in my custom view. Copied method body from current  sqladmin/sqladmin/models.py.

And then just replaced 
```
       search = request.query_params.get("search", None)
```
with
```
       search = request.query_params.get("search", None)
       if search and search != "" and page > 1:
            page = 1
```
This resets current page to first whenever you are trying to search something. If you typically have more than one page of search results then this solution wouldn't have much value for you.
